diesel/src/macros/query_id.rs

```diff
@@ -1,5 +1,36 @@
+/// Provides a standard implementation of `QueryId`.
+/// Apps should not need to concern themselves with this macro.
+///
+/// This macro should be called with the name of your type, along with all of
+/// it's type parameters.
+/// If the SQL generated by your type not uniquely identifiable by the type
+/// itself, you should put `noop:` in front.
+///
+/// For example, given the type `And<Left, Right>`, invoking
+/// `impl_query_id!(And<Left, Right>)` will generate:
+///
+/// ```rust,ignore
+/// impl<Left, Right> QueryId for And<Left, Right>
+/// where
+///     Left: QueryId,
+///     Right: QueryId,
+/// {
+///     type QueryId = And<Left::QueryId, Right::QueryId>;
+///
+///     const HAS_STATIC_QUERY_ID: bool = Left::HAS_STATIC_QUERY_ID && Right::HAS_STATIC_QUERY_ID;
+/// }
+/// ```
+///
+/// Invoking `impl_query_id!(noop: And<Left, Right>)` will generate:
+///
+/// ```rust,ignore
+/// impl<Left, Right> QueryId for And<Left, Right> {
+///     type QueryId = ();
+///
+///     const HAS_STATIC_QUERY_ID: bool = false;
+/// }
+/// ```
 #[macro_export]
-#[doc(hidden)]
```

Comment:
If we want to replace this macro with a custom derive in 1.1 we should hide the documentation about it, to "remove" the item from the public api.

Reply:
We can deprecate the macro in 1.1.